src/InboxMonitor.vala

```diff
@@ -0,0 +1,233 @@
+    }
+
+    private void inbox_folder_changed (E.Source source, Camel.FolderChangeInfo changes) {
+        var inbox_folder = inbox_folders.get (source);
+        if (inbox_folder == null) {
+            return;
+        }
+
+        unowned var added_uids = changes.get_added_uids ();
+        if (added_uids != null) {
+            var sender_names = new GenericSet<string> (str_hash, str_equal);
+            var unseen_message_infos = new SList<Camel.MessageInfo> ();
+
+            added_uids.foreach ((added_uid) => {
+                var message_info = inbox_folder.get_message_info (added_uid);
+
+                if (!(Camel.MessageFlags.SEEN in message_info.flags)) {
+                    unowned string? sender_address;
+                    unowned string? sender_name;
+
+                    var camel_address = new Camel.InternetAddress ();
+                    camel_address.unformat (message_info.from);
+                    camel_address.get (0, out sender_name, out sender_address);
+
+                    if (sender_name == null) {
+                        sender_name = sender_address;
+                    }
+
+                    sender_names.add (sender_name);
+                    unseen_message_infos.append (message_info);
+                }
+            });
+
+            var unseen_message_infos_length = unseen_message_infos.length ();
+            if (unseen_message_infos_length == 1) {
+                var unseen_message_info = unseen_message_infos.nth_data (0);
+
+                var notification = new GLib.Notification (sender_names.iterator ().next_value ());
+                notification.set_body (unseen_message_info.subject);
+                GLib.Application.get_default ().send_notification (unseen_message_info.uid, notification);
+
+            } else if (unseen_message_infos_length > 1) {
+                GLib.Notification notification;
+                string messages_count = ngettext ("%u new message", "%u new messages", unseen_message_infos_length).printf (unseen_message_infos_length);
+
+                if (sender_names.length == 1) {
+                    ///TRANSLATORS: The first %s represents the number of new messages translated in your language, e.g. "2 new messages"
+                    ///The next %s represents the sender name
+                    notification = new GLib.Notification (_("%s from %s").printf (messages_count, sender_names.iterator ().next_value ()));
+                } else {
+                    ///TRANSLATORS: The first %s represents the number of new messages translated in your language, e.g. "2 new messages"
+                    ///The next %s represents the number of senders
+                    notification = new GLib.Notification (ngettext ("%s from %u sender", "%s from %u senders", sender_names.length).printf (messages_count, sender_names.length));
+                }
+
+                GLib.Application.get_default ().send_notification (unseen_message_infos.nth_data (0).uid, notification);
```

Comment:
Does the UID here represent the ID of the sender or of the account?

Reply:
This uid is whether from the sender nor the account: its from the first message received in this batch.

Reply:
Ah okay. In that case, I imagine we should use the ID of the associated account so that when we get replaces working in the notification center it'll show only one up-to-date notification per account instead of having like "10 new messages" and "9 new messages" from the same account saved